### PR TITLE
feat(plus): add method to mount raw volume to a container

### PR DESCRIPTION
**Pull request by abierbaum**
This is the best way I have found to use kplus and allow mounting to all the various k8s volume types.

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/cdk8s-plus/src/volume.ts [hunk 1]
```diff
@@ -98,6 +98,20 @@ export class Volume {
...
+   * Create a volume the wraps a k8s.Volume for usage in cdk8s-plus.
+   * 
+   * @param name Name of the volume.
+   * @param volume The k8s.Volume resource to wrap.
+   * 
+   * note: 
+   *    Using Omit outputs: 
+   *      jsii/compiler - Type model errors prevented the JSII assembly from being created
+   */
+  public static fromVolumeSpec(name: string, volume: any /*Omit<k8s.Volume, 'name'>*/): Volume {
```
Comment (abierbaum) [inline packages/cdk8s-plus/src/volume.ts:111]: I would love to find a way to use Omit here but it causes an error with the jsii/compiler:

```
jsii/compiler - Type model errors prevented the JSII assembly from being created
```

Any ideas?

**Comment by abierbaum** [comment]
@iliapolo Any thoughts on this?  If it isn't something the team wants, I can extract it to an addon in my local project.  Just want to know if I should close it.

**Comment by iliapolo** [comment]
@abierbaum Sorry for the delay. 

I understand the use case of this PR, but we deliberately try not to expose the underlying L1 spec. 
So I feel this needs a different approach.

I'm ok with leaving it open for now just so we will have it on the radar for later. Added the `needs-discussion` label to signal that.

**Comment by abierbaum** [comment]
@iliapolo I don't disagree that it is ugly and should be improved.  In the meantime, what is your recommended way to do something like add a host volume.  I only added this path because I couldn't see another way to do it.

**Comment by iliapolo** [comment]
Hi @abierbaum - Apologies for the delayed response. I think the best option right now is to use escape hatches, which we've just recently added:

https://cdk8s.io/docs/v1.0.0-beta.3/concepts/escape-hatches/

Going forward i think we should add `Volume.fromHostPath` API, similar to like the ones we already have. WDYT? 

Thanks

**Comment by abierbaum** [comment]
> Hi @abierbaum - Apologies for the delayed response. I think the best option right now is to use escape hatches, which we've just recently added:
> 
> https://cdk8s.io/docs/v1.0.0-beta.3/concepts/escape-hatches/
> 
> Going forward i think we should add `Volume.fromHostPath` API, similar to like the ones we already have. WDYT?
> 
> Thanks

Seems reasonable.  One thing I would suggest considering is exposing the Volume constructor to allow more directly use when requested.  The JSONPatch support could be used, but seems like it would be a bit of a pain to do things that are abnormal.

For my cases I just created a custom construct that extends Volume and supports the use cases I need.

**Pull request closed without merging**